### PR TITLE
FIPS: Remove pkispawn cruft

### PR DESCRIPTION
**Pull request by stlaz**
`pkispawn` leaves some ugly files after its successful run. This patch:
a) makes sure the files are removed (say no to `__del__` in `DogtagInstance`)
b) prevents special requirements for DM password in FIPS as this was for some reason used to create an NSS database for `pkispawn`

### CODE REVIEW

**Comment by MartinBasti** [comment]
```
************* Module ipaserver.install.cainstance
ipaserver/install/cainstance.py:685: [E1101(no-member), CAInstance.import_ra_cert] Instance of 'CAInstance' has no 'ra_agent_db' member)
ipaserver/install/cainstance.py:685: [E1101(no-member), CAInstance.import_ra_cert] Instance of 'CAInstance' has no 'ra_agent_pwd' member)
ipaserver/install/cainstance.py:831: [E1101(no-member), CAInstance.__request_ra_certificate] Instance of 'CAInstance' has no 'ra_agent_db' member)
ipaserver/install/cainstance.py:834: [E1101(no-member), CAInstance.__request_ra_certificate] Instance of 'CAInstance' has no 'ra_agent_pwd' member)
************* Module ipaserver.install.dogtaginstance
ipaserver/install/dogtaginstance.py:78: [E0602(undefined-variable), export_kra_agent_pem] Undefined variable 'tempfile')
```

File: ipaserver/install/dogtaginstance.py [hunk 1]
```diff
@@ -171,6 +167,15 @@ def spawn_instance(self, cfg_file, nolog_list=()):
         except ipautil.CalledProcessError as e:
             self.handle_setup_error(e)
 
+    def _clean_pkispawn_files(self):
+        if self.tmp_agent_db is None:
```
Comment (tiran) [inline ipaserver/install/dogtaginstance.py]: You are skipping the removal of ```/root/.dogtag/pki-tomcat/{subsystem}```, too.

**Comment by stlaz** [comment]
Hm, originally had this over the nsslib removal patchset but the rebase was not as successful as I thought, will fix the issues.

File: ipaserver/install/dogtaginstance.py (hunk 1 above)
Comment (stlaz) [inline ipaserver/install/dogtaginstance.py]: Right, it was my impression that this one would not exist if `pkispawn`was not run, I can just remove this check and run removal of both trees.

**Comment by tiran** [comment]
Or you could always clean up ```/root/.dogtag``` and remove the tmp dir when the var is not None.

By the way do you clean up ```/root/.dogtag``` during update?

**Comment by stlaz** [comment]
Always tend to forget about the upgrade part, will do, thanks 👍

**Comment by tiran** [comment]
pylint needs some attention, too.

```
************* Module ipaserver.install.cainstance
ipaserver/install/cainstance.py:685: [E1101(no-member), CAInstance.import_ra_cert] Instance of 'CAInstance' has no 'ra_agent_db' member)
ipaserver/install/cainstance.py:685: [E1101(no-member), CAInstance.import_ra_cert] Instance of 'CAInstance' has no 'ra_agent_pwd' member)
ipaserver/install/cainstance.py:831: [E1101(no-member), CAInstance.__request_ra_certificate] Instance of 'CAInstance' has no 'ra_agent_db' member)
ipaserver/install/cainstance.py:834: [E1101(no-member), CAInstance.__request_ra_certificate] Instance of 'CAInstance' has no 'ra_agent_pwd' member)
************* Module ipaserver.install.dogtaginstance
ipaserver/install/dogtaginstance.py:78: [E0602(undefined-variable), export_kra_agent_pem] Undefined variable 'tempfile')
```

**Comment by stlaz** [comment]
All should be fixed now.

**Comment by tomaskrizek** [comment]
Works as expected.

**Comment by pvoborni** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/728a6bd4229ba170b2e94f216127b19d5d94e2ba
https://fedorahosted.org/freeipa/changeset/a39effed7603d66acd238e3142f4df8081ff7bc8

**Pull request closed without merging**